### PR TITLE
 implementation of Dijkstra's algorithm in Java

### PR DESCRIPTION
**Pull request by whitehawk0910**

### CODE REVIEW

File: dijkstra.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import java.util.Arrays;
+
+public class DijkstraAlgorithm {
```
Comment (rohan472000) [inline dijkstra.java:3]: Add description about this algo with link....
Also add comments in every function or wherever required.

File: dijkstra.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    private static final int INFINITY = Integer.MAX_VALUE;
+
+    public static void main(String[] args) {
+        int[][] graph = {
```
Comment (rohan472000) [inline dijkstra.java:8]: Add more unique cases

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Comment by whitehawk0910** [comment]
yes , i made the changes as you instructed

**Review by rohan472000 — CHANGES_REQUESTED** [review]
Add more unique testcases in main.

**Comment by SAURABHTIWARIVBSPU** [comment]
import java.io.*;
import java.util.*;
public class Main {
	
	  // making an edge class that stores the vertex and its neighbor along with the weight
	  static class Edge {
	    int source;
	    int nbr;
	    int weight;
	 
	    Edge(int source, int nbr, int weight) {
	      this.source = source;
	      this.nbr = nbr;
	      this.weight = weight;
	    }
	  }
	  
	  //making a pair class
	 static class Pair implements Comparable< Pair> {
	    int vertex;
	    int wsf; //weight so far
	 
	    Pair(int vertex, int wsf) {
	      this.vertex = vertex;
	      this.wsf = wsf;
	    }
	    // using CompareTo function so that priority queue arranges the element in terms of weight so far
	    public int compareTo(Pair o) {
	      return this.wsf - o.wsf;
	    }
	  }
	  
	  public static void main(String[] args) {
	    
	    // building the graph
		// the graph is an array of ArrayList (of Object,Edge)
		// length of array = the vertices present in graph.
		  
	    ArrayList<Edge> [] graph =  new ArrayList[7];
	    for(int i = 0; i < graph.length; i++){
	        graph[i] = new ArrayList<>();
	    }
	    
	    // Adding all the vertices and its members along with weights
	    
	    graph[0].add(new Edge(0,1,14));
	    graph[0].add(new Edge(0,3,5));
	    
	    graph[1].add(new Edge(1,0,14));
	    graph[1].add(new Edge(1,4,2));
	    graph[1].add(new Edge(1,2,6));
	    
	    graph[2].add(new Edge(2,1,6));
	    graph[2].add(new Edge(2,5,1));
	    graph[2].add(new Edge(2,6,7));
	    
	    graph[3].add(new Edge(3,0,5));
	    graph[3].add(new Edge(3,4,3));
	    
	    graph[4].add(new Edge(4,3,3));
	    graph[4].add(new Edge(4,1,2));
	    
	    graph[5].add(new Edge(5,2,1));
	    graph[5].add(new Edge(5,6,8));
	    
	    graph[6].add(new Edge(6,2,7));
	    graph[6].add(new Edge(6,5,8));
	    
	    //taking 0 as source node
	    int src = 0;
	    
	    //creating a visited boolean array
	    boolean[] visited = new boolean[graph.length];
	    
	    //Initialising a priority Queue of object Pair
	    PriorityQueue< Pair> prioQue = new PriorityQueue<>();
	    prioQue.add(new Pair(src, 0));
	    
	    // while queue is not empty following the procedure remove,mark,print,add
	    
	    while (prioQue.size() > 0) {
	    	
	      // removing the topmost element and storing it in topEle
	      Pair topEle = prioQue.remove();
	      
	      //checking if vertex already visited
	      if (visited[topEle.vertex] == true) {
	        continue;
	      }
	      // marking true if  vertex not visited
	      visited[topEle.vertex] = true;
	      
	      //Printing the vertex and weight to reach that vertex
	      System.out.println("Vertex :" + " " +topEle.vertex + " & " + "Weight so far :" + " " +topEle.wsf);
	      
	      //adding all the unvisited neighbor of vertex in queue
	      for (Edge edge : graph[topEle.vertex]) {
	        if (visited[edge.nbr] == false) {
	          prioQue.add(new Pair(edge.nbr, topEle.wsf + edge.weight ));
	        }
	      }
	    }
	  }

**Review by SAURABHTIWARIVBSPU — CHANGES_REQUESTED** [review]
// making an edge class that stores the vertex and its neighbor along with the weight
  static class Edge {
    int source;
    int nbr;
    int weight;
 
    Edge(int source, int nbr, int weight) {
      this.source = source;
      this.nbr = nbr;
      this.weight = weight;
    }
  }
  
  //making a pair class
 static class Pair implements Comparable< Pair> {
    int vertex;
    int wsf; //weight so far
 
    Pair(int vertex, int wsf) {
      this.vertex = vertex;
      this.wsf = wsf;
    }
    // using CompareTo function so that priority queue arranges the element in terms of weight so far
    public int compareTo(Pair o) {
      return this.wsf - o.wsf;
    }
  }
  
  public static void main(String[] args) {
    
    // building the graph
	// the graph is an array of ArrayList (of Object,Edge)
	// length of array = the vertices present in graph.
	  
    ArrayList<Edge> [] graph =  new ArrayList[7];
    for(int i = 0; i < graph.length; i++){
        graph[i] = new ArrayList<>();
    }
    
    // Adding all the vertices and its members along with weights
    
    graph[0].add(new Edge(0,1,14));
    graph[0].add(new Edge(0,3,5));
    
    graph[1].add(new Edge(1,0,14));
    graph[1].add(new Edge(1,4,2));
    graph[1].add(new Edge(1,2,6));
    
    graph[2].add(new Edge(2,1,6));
    graph[2].add(new Edge(2,5,1));
    graph[2].add(new Edge(2,6,7));
    
    graph[3].add(new Edge(3,0,5));
    graph[3].add(new Edge(3,4,3));
    
    graph[4].add(new Edge(4,3,3));
    graph[4].add(new Edge(4,1,2));
    
    graph[5].add(new Edge(5,2,1));
    graph[5].add(new Edge(5,6,8));
    
    graph[6].add(new Edge(6,2,7));
    graph[6].add(new Edge(6,5,8));
    
    //taking 0 as source node
    int src = 0;
    
    //creating a visited boolean array
    boolean[] visited = new boolean[graph.length];
    
    //Initialising a priority Queue of object Pair
    PriorityQueue< Pair> prioQue = new PriorityQueue<>();
    prioQue.add(new Pair(src, 0));
    
    // while queue is not empty following the procedure remove,mark,print,add
    
    while (prioQue.size() > 0) {
    	
      // removing the topmost element and storing it in topEle
      Pair topEle = prioQue.remove();
      
      //checking if vertex already visited
      if (visited[topEle.vertex] == true) {
        continue;
      }
      // marking true if  vertex not visited
      visited[topEle.vertex] = true;
      
      //Printing the vertex and weight to reach that vertex
      System.out.println("Vertex :" + " " +topEle.vertex + " & " + "Weight so far :" + " " +topEle.wsf);
      
      //adding all the unvisited neighbor of vertex in queue
      for (Edge edge : graph[topEle.vertex]) {
        if (visited[edge.nbr] == false) {
          prioQue.add(new Pair(edge.nbr, topEle.wsf + edge.weight ));
        }
      }
    }
  }

**Comment by rohan472000** [comment]
@SAURABHTIWARIVBSPU , appreciate your enthusiasm but this is not a way to review by pasting code at once...you should point out where exactly one needs to change or improve something, give suggestions as there are options available in review.

Better you should learn how to contribute and review, take help of YouTube.

**Comment by whitehawk0910** [comment]
addded more test cases

**Review by rohan472000 — APPROVED** [review]

**Comment by rohan472000** [comment]
Thanks for your contribution.

**Pull request merged**